### PR TITLE
Cleaned up dependencies for metacat-client and metacat-common.

### PR DESCRIPTION
**Pull request by ajoymajumdar**
All transitive dependencies except metacat-common are relocated in the shaded jar.

#1. Upgraded gradle to 5.4.1
#2. Cleaned up unwanted dependencies.
#3. Added functionalTestShaded to test the shaded jar.

### CODE REVIEW

**Comment by zhljen** [comment]
Wondering if we should create this shade client as internal metacat, which will be easier to maintain.

**Comment by ajoymajumdar** [comment]
Need to keep the binaries along with its other jar. This way you maintain the version. We are shading all transitive dependencies.

File: metacat-functional-tests/src/functionalTestShaded/groovy/com/netflix/metacat/MetacatSmokeSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,1467 @@
...
+import java.util.concurrent.TimeUnit
+
+/**
+ * MetacatSmokeSpec.
+ * @author amajumdar
+ * @author zhenl
+ * @since 1.0.0
+ */
+class MetacatSmokeSpec extends Specification {
+    public static MetacatV1 api
```
Comment (zhljen) [inline metacat-functional-tests/src/functionalTestShaded/groovy/com/netflix/metacat/MetacatSmokeSpec.groovy]: Are these tests are the same as the existing functional test except for testing shaded jar?

File: metacat-functional-tests/build.gradle [hunk 2]
```diff
@@ -63,18 +67,20 @@ dependencies {
      *******************************/
 
     // Test that metacat-client can be used with jersey 1
-    testCompile(project(":metacat-client")) {
-        exclude group: "javax.ws.rs", module: "javax.ws.rs-api"
-        exclude group: "org.glassfish.jersey.core"
-    }
+    functionalTestShadedCompile project(path: ':metacat-client', configuration: 'shadow')
+
```
Comment (zhljen) [inline metacat-functional-tests/build.gradle]: I don't understand this 'Test that metacat-client can be used with jersey 1'..

**Comment by zhljen** [comment]
I still wonder if we can consider pulling back the fat client internally, maybe as a separate project.

File: metacat-client/build.gradle [hunk 3]
```diff
@@ -56,3 +67,26 @@ dependencies {
 
     testCompile(project(":metacat-common").sourceSets.test.output)
 }
+
+// All transitive dependencies are relocated except metacat-common.
+shadowJar {
```
[the remark this answers is not in the document]
Comment (ajoymajumdar) [reply metacat-client/build.gradle]: 5.3MB. non-shaded jar is 19KB

File: metacat-functional-tests/src/functionalTestShaded/groovy/com/netflix/metacat/MetacatSmokeSpec.groovy (hunk 1 above)
Comment (ajoymajumdar) [reply metacat-functional-tests/src/functionalTestShaded/groovy/com/netflix/metacat/MetacatSmokeSpec.groovy]: Right.

File: metacat-functional-tests/build.gradle (hunk 2 above)
Comment (ajoymajumdar) [reply metacat-functional-tests/build.gradle]: This was an old comment. Then the platform supported jersey 1. We can remove the comment and dependency in the test.

Comment (danielbwatson) [reply metacat-functional-tests/build.gradle]: That is really old, from before Metacat moved to Spring.  When we were transitioning to Java 8 I added a test to make sure Metacat worked with both Jersey 1 and 2.  It looks like only the comment remains and can be safely removed.

**Pull request merged**